lisa/sut_orchestrator/azure/features.py

```diff
@@ -2397,6 +2397,15 @@ def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
                 "Or consider changing the disk type or location."
             )
 
+        # If maximize_capability is set, only use AvailabilitySet if
```

Comment:
The logic doesn't check for max cap, how to make sure it applied only max cap is set?

Reply:
If we enter the 'not max cap' logic above, there is no way for the availability_type to still be default.

Reply:
Explicitly logic is clearer, if change above condition to `if is_maximize_capability`, and put the new logic there.

Reply:
Maybe it would be better if I changed the comment?
I wrote the code in a way such that if for any reason the value is still default when it shouldn't be, then resolve the default.